frontend/packages/kubevirt-plugin/src/components/vm-templates/vm-template-source.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import { TemplateKind } from '@console/internal/module/k8s';
+import { Tooltip } from '@patternfly/react-core';
+import { parseURL, resolveURL } from '../../utils/url';
+import { ProvisionSource } from '../../constants/vm/provision-source';
+
+export const URLObj: React.FC<URLObjProps> = ({
+  urlPath,
+  short = false,
+  maxHostnameParts = 3,
+  maxPathnameParts = 1,
+}) => {
+  const urlObj = short ? parseURL(urlPath) : undefined;
+
+  const resolvedURL = urlObj ? resolveURL({ urlObj, maxHostnameParts, maxPathnameParts }) : urlPath;
+  const resolvedTitle = resolvedURL === urlPath ? undefined : urlPath;
+
+  return (
+    <a href={urlPath} title={resolvedTitle}>
+      {resolvedURL}
+    </a>
+  );
+};
+
+type URLObjProps = {
+  urlPath: string;
+  short?: boolean;
+  maxHostnameParts?: number;
+  maxPathnameParts?: number;
+};
+
+const Type: React.FC<TypeProps> = ({ type, source, error, detailed }) => {
+  if (error || !type) {
+    if (!detailed) {
+      return (
+        <Tooltip position="bottom" enableFlip={false} content={<>{error}</>} exitDelay={0}>
+          <div>Invalid source</div>
```

Comment:
something like this.. + different `ifs` above
```suggestion
         <div>{error ? 'Invalid Source' : type}</div>
```